### PR TITLE
Refactor Connect Inject Webhook to use webhook-cert-manager

### PR DESCRIPTION
**Pull request by thisisnotashwin**
This PR is a partner to hashicorp/consul-k8s#454 and proposes the following changes:
- Enable webhook-cert-manager whenever either controller or connectInject is enabled
- Remove connectInject.certs values. It seems like this behavior was already broken, and we don't want to support it going forward with webhook-cert-manager.

The draft was created by @thisisnotashwin and cleanup and tests added by @ishustava.

---
[original description]
This is a rough draft but is at a good place to see the changes made and the impact on Helm. This will eventually be a PR against a `feature-tproxy` branch

### CODE REVIEW

**Comment by lkysow** [comment]
@thisisnotashwin lmk when you want a full review, this approach looks good to me.

File: templates/connect-inject-deployment.yaml [hunk 1]
```diff
@@ -185,81 +181,46 @@ spec:
...
-            successThreshold: 1
-            timeoutSeconds: 5
-          startupProbe:
-            httpGet:
-              path: /health/ready
-              port: 8080
-              scheme: HTTPS
-            failureThreshold: 15
-            periodSeconds: 2
-            timeoutSeconds: 5
```
Comment (ishustava) [inline templates/connect-inject-deployment.yaml:215]: we're removing these for the time being as we'd need to address them differently now that webhook runs through the operator-sdk. It'll be addressed in a separate PR.

**Review by lkysow — APPROVED** [review]
code reviewed only

**Review by ndhanushkodi — APPROVED** [review]
Just had some questions more for my own understanding for things I was confused about. Otherwise, the rest of it looks good to me!!

File: templates/connect-inject-mutatingwebhook.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ webhooks:
         name: {{ template "consul.fullname" . }}-connect-injector-svc
         namespace: {{ .Release.Namespace }}
         path: "/mutate"
-      caBundle: {{ .Values.connectInject.certs.caBundle | quote }}
+      caBundle: Cg==
```
[the remark this answers is not in the document]
Comment (ishustava) [reply templates/connect-inject-mutatingwebhook.yaml:29]: yeah it's a base64 encoded newline. Will add!

[the remark this answers is not in the document]
Comment (ishustava) [reply templates/connect-inject-mutatingwebhook.yaml]: Acutally, I realized that this property is not required, so I removed it instead to avoid confusion.
